### PR TITLE
WELD-2430 Throw EJBException when trying to invoke non-public method …

### PR DESCRIPTION
**Pull request by manovotn**
…on EJB bean.

### CODE REVIEW

**Comment by manovotn** [comment]
To keep the confusion to minimum I basically copied over the exception coming from WFLY. With the test (assuming you remove the catch block) you now get: `WELD-001573: Not a business method java.lang.String org.jboss.weld.tests.ejb.stateless.noInterfaceNonPublic.ProblematicBean.ping(). Do not call non-public methods on EJBs.`

File: impl/src/main/java/org/jboss/weld/logging/BeanLogger.java [hunk 1]
```diff
@@ -505,4 +507,7 @@
 
     @Message(id = 1572, value = "Cannot create instance of session bean from Annotated Type {0} before AfterDeploymentValidation phase.", format = Format.MESSAGE_FORMAT)
     CreationException initABDnotInvoked(Object bean);
+
+    @Message(id = 1573, value = "Not a business method {0}. Do not call non-public methods on EJB's.", format = Format.MESSAGE_FORMAT)
+    EJBException nonPublicMethodInvoked(Object method);
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/logging/BeanLogger.java]: Note that EJB API is an optional dependency and `BeanLogger` is used in SE and other environments. So I think we should not import this class here.

Comment (manovotn) [reply impl/src/main/java/org/jboss/weld/logging/BeanLogger.java]: Agreed, moving this to `EnterpriseBeanProxyMethodHandler`

**Comment by mkouba** [comment]
Looks good.

**Pull request merged**